res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -1241,57 +1028,35 @@ NumarkMixtrack3.initDeck = function(group, remove) {
 
 NumarkMixtrack3.connectDeckControls = function(group, remove) {
     // If the 'remove' parameter is not passed to this function, set remove = false
-    remove = (typeof remove !== 'undefined') ? remove : false;
-    var OnDeck = parseInt(NumarkMixtrack3.channelRegEx.exec(group)[1]); 
-    var OffDeck = OnDeck;
-    var i; 
-
-    if (OffDeck <= 2) {
-        OffDeck += 2;
-    } else {
-        OffDeck -= 2;
-    }
+    remove = remove || false;
+    var onDeckNum = parseInt(NumarkMixtrack3.channelRegEx.exec(group)[1]); 
+    var offDeckNum = (onDeckNum + 1) % 4 + 1;
+    var onDeck = NumarkMixtrack3.decks["D" + onDeckNum];
+    var offDeck = NumarkMixtrack3.decks["D" + offDeckNum];
 
     if (remove) {
         print("==========================================================");
-        print("           Disconnect controls from deck " + OffDeck);
+        print("           Disconnect controls from deck " + offDeckNum);
         print("");
 
         // make sure that the shift is no longer active on either deck to prevent confusion
-        NumarkMixtrack3.decks["D" + OffDeck].shiftKey = false;
-        NumarkMixtrack3.decks["D" + OnDeck].shiftKey = false; 
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.jogWheelsInScratchMode.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.headphones.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.loopin.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.loopout.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.reloop_exit.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.loop_halve.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.hotCue1.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.hotCue2.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.hotCue3.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.hotCue4.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.Cue.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.sync.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.play.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.fx1.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.fx2.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.fx3.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.tap.onOff(OFF);
-        NumarkMixtrack3.decks["D" + OffDeck].LEDs.meter.onOff(OFF);
-        
-        for (i = 0; i < 4; i++) {
-        var index = i + 1;
+        offDeck.shiftKey = false;
+        onDeck.shiftKey = false; 
+
+        for (var led in offDeck.LEDs) {
+            if (led.hasOwnProperty('onOff')) {
+                led.onOff(OFF);
```

Comment:
I'm not too familiar with this code but I agree. However, I think that can be part of a future refactor of the LED/Button objects. Either that or leave it and consider using the components lib you developed.